src/uct/ib/rc/base/rc_iface.h

```diff
@@ -198,6 +198,7 @@ struct uct_rc_iface {
         ucs_mpool_t             mp;       /* pool for send descriptors */
         ucs_mpool_t             fc_mp;    /* pool for FC grant pending requests */
         ucs_mpool_t             flush_mp; /* pool for flush completions */
+        ucs_mpool_t             fence_mp; /* pool for fence completions */
```

Comment:
maybe reuse fc_mp, which should be relatively small

Reply:
fc_mp is initialized by init_attr->fc_req_size element size, not sure it is ```relatively small```

Reply:
Can be checked. Most probably using one memory pool would be better in terms of memory consumption (we already have serveral different).. wdyt?

Reply:
fc_mp size depends from configuration (runtime), it is really bad candidate to union, flush_mp - is too huge to merge. additional mem pool is just one more malloc per iface, but it allows to simplify logic

Reply:
Ok.
i did not mean additional malloc, we will have to allocate one more chunk of elements which will increase default memory consumption